system-tests/tests/smoke/cre/workflow_don_load_test.go

```diff
@@ -0,0 +1,915 @@
+			}
+			if flags.HasFlag(donWithMetadata.Flags, keystonetypes.OCR3Capability) {
+				for i := range feedsAddresses {
+					feedConfig := make([]FeedConfig, 0)
+
+					for _, feed := range feedsAddresses[i] {
+						feedID, err2 := datastreams.NewFeedID(feed.Feed)
+						if err2 != nil {
+							return nil, err2
+						}
+						feedBytes := feedID.Bytes()
+						feedConfig = append(feedConfig, FeedConfig{
+							FeedIDsIndex: int(feed.StreamID),
+							Deviation:    "0.001",
+							Heartbeat:    3600,
+							RemappedID:   "0x" + hex.EncodeToString(feedBytes[:]),
+						})
+					}
+
+					jobSpec := TextWorkflow(nodeID, fmt.Sprintf("load_%d", i), feedConfig)
+					jobDesc := keystonetypes.JobDescription{Flag: keystonetypes.OCR3Capability, NodeType: keystonetypes.WorkerNode}
+
+					if _, ok := jobSpecs[jobDesc]; !ok {
+						jobSpecs[jobDesc] = []*jobv1.ProposeJobRequest{jobSpec}
+					} else {
+						jobSpecs[jobDesc] = append(jobSpecs[jobDesc], jobSpec)
+					}
+				}
+			}
+
+			if flags.HasFlag(donWithMetadata.Flags, keystonetypes.MockCapability) && in.MockCapabilities != nil {
+				jobSpec := MockCapabilitiesJob(nodeID, in.MockCapabilities)
+				jobDesc := keystonetypes.JobDescription{Flag: keystonetypes.MockCapability, NodeType: keystonetypes.WorkerNode}
+
+				if _, ok := jobSpecs[jobDesc]; !ok {
+					jobSpecs[jobDesc] = []*jobv1.ProposeJobRequest{jobSpec}
+				} else {
+					jobSpecs[jobDesc] = append(jobSpecs[jobDesc], jobSpec)
+				}
+			}
+		}
+		return jobSpecs, nil
+	}
+
+	WorkflowDONLoadTestCapanilitiesFactoryFn := func(donFlags []string) []keystone_changeset.DONCapabilityWithConfig {
+		var capabilities []keystone_changeset.DONCapabilityWithConfig
+
+		if flags.HasFlag(donFlags, keystonetypes.MockCapability) {
+			for _, m := range in.MockCapabilities {
+				capabilities = append(capabilities, keystone_changeset.DONCapabilityWithConfig{
+					Capability: kcr.CapabilitiesRegistryCapability{
+						LabelledName:   m.Name,
+						Version:        m.Version,
+						CapabilityType: capTypeToInt(m.Type),
+					},
+					Config: &capabilitiespb.CapabilityConfig{},
+				})
+			}
+		}
+
+		if flags.HasFlag(donFlags, keystonetypes.CustomComputeCapability) {
+			capabilities = append(capabilities, keystone_changeset.DONCapabilityWithConfig{
+				Capability: kcr.CapabilitiesRegistryCapability{
+					LabelledName:   "custom-compute",
+					Version:        "1.0.0",
+					CapabilityType: 1, // ACTION
+				},
+				Config: &capabilitiespb.CapabilityConfig{},
+			})
+		}
+
+		if flags.HasFlag(donFlags, keystonetypes.OCR3Capability) {
+			capabilities = append(capabilities, keystone_changeset.DONCapabilityWithConfig{
+				Capability: kcr.CapabilitiesRegistryCapability{
+					LabelledName:   "offchain_reporting",
+					Version:        "1.0.0",
+					CapabilityType: 2, // CONSENSUS
+					ResponseType:   0, // REPORT
+				},
+				Config: &capabilitiespb.CapabilityConfig{},
+			})
+		}
```

Comment:
curious why we need these two as well? you could pass `WorkflowDONLoadTestCapanilitiesFactoryFn` together with the default factory function that knows how to register OCR and custom compute